### PR TITLE
simplify js build steps

### LINKED ISSUE #1137: setup js does not reliably install webpack
Webpack was recently added as a development dependency, but when I went to run `setup js`, it did not get installed.

Looking at `setupbase.py`, it appears that it only considers running `npm install` if it's also going to run bower. I suspect this is wrong, but I don't know how these pieces fit together.

### PR DESCRIPTION
**Pull request by minrk**
- Run npm, bower install every time (except from sdist).
  Removes need to check npm, bower sources.
  These commands aren't slow enough that we need to save this time (~1s).
- Only check `built/index.js` as build target,
  since webpack is only a single step now.

closes #1137

### CODE REVIEW

**Comment by blink1073** [comment]
:+1:!

**Comment by takluyver** [comment]
Travis failures:

```
      File "/tmp/pip-build-zy691q2m/notebook/setupbase.py", line 506, in run
        targets.extend(js.targets)
      File "/opt/python/3.5.1/lib/python3.5/distutils/cmd.py", line 103, in __getattr__
        raise AttributeError(attr)
    AttributeError: targets
```

File: setupbase.py [hunk 1]
```diff
@@ -326,8 +326,8 @@ def run(cmd, *args, **kwargs):
     return check_call(cmd, *args, **kwargs)
 
 
-class Bower(Command):
-    description = "fetch static client-side components with bower"
+class JavascriptDependencies(Command):
+    description = "fetch static client-side components with npm and bower"
```
Comment (takluyver) [inline setupbase.py]: It's also fetching JS build tools like less and webpack, right?

Comment (jdfreder) [inline setupbase.py]: yup

**Comment by jdfreder** [comment]
Thanks!  This is great!

File: setupbase.py (hunk 1 above)
Comment (minrk) [inline setupbase.py]: Yup, changed to be more generally accurate "Javascript dependencies"

File: setupbase.py [hunk 2]
```diff
@@ -444,56 +420,39 @@ def initialize_options(self):
...
+        if self.force or not os.path.exists(self.target):
+            print("Missing %s" % self.target)
             return True
-        target_mtime = mtime(target)
-        for source in self.sources(name):
+        target_mtime = mtime(self.target)
+        for source in self.sources():
             if mtime(source) > target_mtime:
-                print(source, target)
+                print('%s > %s' % (source, self.target))
```
Comment (takluyver) [inline setupbase.py]: Should this be using distutils' logging machinery, at a level that's suppressed by default? I could imagine this getting very verbose.

Comment (takluyver) [inline setupbase.py]: Nvm, just spotted that it will only print once, because of the short circuit return.

**Pull request merged**